src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -78,19 +101,19 @@ public void init(Properties properties, String id) {
         // Load Configurations
         HashiCorpVaultConfigLoader hashiCorpVaultConfigLoader = HashiCorpVaultConfigLoader.getInstance();
         try {
-            address = hashiCorpVaultConfigLoader.getProperty(ADDRESS_PARAMETER);
-            namespace = hashiCorpVaultConfigLoader.getProperty(NAMESPACE_PARAMETER);
-            enginePath = hashiCorpVaultConfigLoader.getProperty(ENGINE_PATH_PARAMETER);
+            address = hashiCorpVaultConfigLoader.getProperty(propertyPrefix + ADDRESS_PARAMETER);
+            namespace = hashiCorpVaultConfigLoader.getProperty(propertyPrefix + NAMESPACE_PARAMETER);
+            enginePath = hashiCorpVaultConfigLoader.getProperty(propertyPrefix + ENGINE_PATH_PARAMETER);
 
-            String version = hashiCorpVaultConfigLoader.getProperty(ENGINE_VERSION_PARAMETER);
+            String version = hashiCorpVaultConfigLoader.getProperty(propertyPrefix + ENGINE_VERSION_PARAMETER);
             engineVersion = version != null ? Integer.parseInt(version) : DEFAULT_ENGINE_VERSION;
+            // Get the vault token
+            retrieveRootToken();
         } catch (HashiCorpVaultException e) {
             LOG.error(e.getMessage(), e);
         }
 
-        // Get the vault token
-        token = System.getenv(TOKEN_PARAMETER);
-        if (token == null || token.isEmpty()) {
+        if (rootToken == null || rootToken.isEmpty()) {
             LOG.warn("VAULT_TOKEN environment variable is not set");
```

Comment:
Update the warning message. As we are no longer taking the token as an env variable.

Reply:
Updated

Reply:
Updated